connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -1138,5 +1142,13 @@ else if (value instanceof Long)
     private String className(Object o) {
         return o != null ? o.getClass().getName() : "null";
     }
-}
 
+    private static void maybeAddClientId(Map<String, ? super String> clientProps, String groupId) {
+        String clientId = "connect-cluster";
+        if (groupId != null) {
```

Comment:
Can `groupId` actually be null here? This should only run in distributed mode

Reply:
Right now we have that guarantee, but since the `*BackingStore` classes accept a `WorkerConfig` instead of a `DistributedConfig`, it's difficult to enforce that guarantee. Plus, if we ever want to use any of these backing stores with standalone mode (or a new mode that doesn't use Kafka's group membership API), it'll make that migration easier.